### ISSUE TITLE
"Cannot select from parameterized type"

### ISSUE THREAD

**Issue by 4ntoine** [web3j/web3j#238]
Let's say you have function that returns `address[]`:

```
function getPledgers(uint featureKey) public
        forFeature(featureKey)
        constant returns (address[])
    {
        FeatureData memory featureData = data[featureKey];
        return featureData.pledgers;
    }
```

The following smart contract wrapper is generated:

```
    public RemoteCall<List<String>> getPledgers(BigInteger featureKey) {
        Function function = new Function("getPledgers", 
                Arrays.<Type>asList(new org.web3j.abi.datatypes.generated.Uint256(featureKey)), 
                Arrays.<TypeReference<?>>asList(new TypeReference<DynamicArray<Address>>() {}));
        return executeRemoteCallSingleValueReturn(function, List<String>.class);
    }
```

For `List<String>.class` you will get an error:
> Cannot select from parameterized type

This can be fixed by replacing `List<String>` to just `List` in both RemoteCall parameter and return statement:
```
public RemoteCall<List> getPledgers(BigInteger featureKey) {
        Function function = new Function("getPledgers", 
                Arrays.<Type>asList(new org.web3j.abi.datatypes.generated.Uint256(featureKey)), 
                Arrays.<TypeReference<?>>asList(new TypeReference<DynamicArray<Address>>() {}));
        return executeRemoteCallSingleValueReturn(function, List.class);
    }
```

 However it would be great to have it out-of-box.

web3j 3.0.2.

**Comment by bluen** [comment]
Also applies to web3j 3.1.1:
```
function getOwners()
		public
		constant
		onlyWallet
		returns (address[])
	{
		return owners;
	}
```
=> 
```
    public RemoteCall<List<String>> getOwners() {
        Function function = new Function("getOwners", 
                Arrays.<Type>asList(), 
                Arrays.<TypeReference<?>>asList(new TypeReference<DynamicArray<Address>>() {}));
        return executeRemoteCallSingleValueReturn(function, List<String>.class);
    }
```

**Comment by gjeanmart** [comment]
+1 (web3j 3.1.1)

**Comment by kevin-smets** [comment]
Still present in 3.2.0. Thanks for the quick workaround.

**Comment by oscarguindzberg** [comment]
+1 (Tried 3.1.0 and 3.2.0)

**Comment by 4ntoine** [comment]
ETA on new release with this code merged?

**Comment by conor10** [comment]
Please can you confirm if this is fixed in web3j 3.3.1? If so, please close this issue.

**Comment by 4ntoine** [comment]
I can confirm it's fixed (in 3.2.0)
